src/bin/s3Purge.ts

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env node
+
+import { Command } from 'commander'
+import * as figlet from 'figlet'
+import { LIB_VERSION } from '../version';
+import { S3Scan, S3_SCAN_HTTPS_AGENT } from '../lib'
+
+const program = new Command()
+console.log(figlet.textSync(`S3 Purge ${LIB_VERSION}`))
+
+program
+    .version(LIB_VERSION)
+    .description(`A tool Provides a writable stream that expects you to write line-delimited S3 keys
+    into it, and performs an S3.deleteObject request on each key`)
+    .usage(`Usage: s3purge <s3url> [--concurrency=num] [--dryrun] [--quiet]`)
+    .argument('<s3Url>', 'an S3 uri of the type s3://bucket/prefix')
+    .option('-d, --dryrun [boolean]', 'Simulate the deletion only', false)
+    .option("-q, --quiet  [boolean]", "Boolean value to run in silent mode", false)
+    .option("-c, --concurrency  [number]", "Concurrency to use")
+    .parse(process.argv);
+
+const options = { agent: S3_SCAN_HTTPS_AGENT, body: true, ...program.opts() }
+const s3Url = program.args[0]
+var interval: NodeJS.Timeout
+const { quiet, dryrun } = program.opts()
+const purge = S3Scan.Purge(s3Url, options, (err) => {
+        if (!quiet) clearInterval(interval)
+        setTimeout(function() {
```

Comment:
I think we should go for a consistent style for all the unnamed functions. Use `() => { }` instead of `function() { }`.

```suggestion
        setTimeout(() => {
```